cpp/models/ide_seir/ide_seir.cpp

```diff
@@ -0,0 +1,141 @@
+    {
+        if ((m_parameters.get<LatencyTime>() < tau) &&
+            (m_parameters.get<InfectiousTime>() + m_parameters.get<LatencyTime>() > tau)) {
+            return tgamma(p + q) * pow(tau - m_parameters.get<LatencyTime>(), p - 1) *
+                   pow(m_parameters.get<InfectiousTime>() + m_parameters.get<LatencyTime>() - tau, q - 1) /
+                   (tgamma(p) * tgamma(q) * pow(m_parameters.get<InfectiousTime>(), p + q - 1));
+        }
+        return 0.0;
+    }
+
+    double IdeSeirModel::central_difference_quotient(TimeSeries<double> const& ts_ide, iSeirInfType compartment,
+                                                     Eigen::Index idx) const
+    {
+        return (ts_ide[idx + 1][Eigen::Index(compartment)] - ts_ide[idx - 1][Eigen::Index(compartment)]) / (2 * m_dt);
+    }
+
+    double IdeSeirModel::num_integration_inner_integral(Eigen::Index idx) const
+    {
+        double res     = 0.5 * (generalized_beta_distribution(m_result.get_time(idx) - m_result.get_time(idx - m_k)) *
+                                central_difference_quotient(m_result, iSeirInfType::S, m_k) +
+                            generalized_beta_distribution(m_result.get_time(idx) - m_result.get_time(idx - m_l)) *
+                                central_difference_quotient(m_result, iSeirInfType::S, idx - m_l));
+        Eigen::Index i = idx - m_k + 1;
+        while (i <= idx - m_l - 2) {
+            res += (generalized_beta_distribution(m_result.get_time(idx) - m_result.get_time(i)) *
+                    central_difference_quotient(m_result, iSeirInfType::S, i));
+            ++i;
+        }
+        return res;
+    }
+
+    TimeSeries<double> const& IdeSeirModel::simulate(int t_max)
+    {
+
+        m_l = (int)std::floor(m_parameters.get<LatencyTime>() / m_dt);
+        m_k = (int)std::ceil((m_parameters.get<InfectiousTime>() + m_parameters.get<LatencyTime>()) / m_dt);
+        if (m_result.get_time(0) > -(m_k - 1) * m_dt) {
+            log_warning("Constraint check: Initial data starts later than necessary. The simulation may be distorted. "
+                        "Start the data at time {:.4f} at the latest.",
+                        -(m_k - 1) * m_dt);
+        }
+
+        while (m_result.get_last_time() < t_max) {
+            m_result.add_time_point(m_result.get_last_time() + m_dt);
+            Eigen::Index idx = m_result.get_num_time_points();
+
+            // R0t is the effective reproduction number at time t
+            auto R0t1 = m_parameters.get<ContactFrequency>().get_cont_freq_mat().get_matrix_at(
+                            m_result.get_time(idx - 2))(0, 0) *
+                        m_parameters.get<TransmissionRisk>() * m_parameters.get<InfectiousTime>();
+            auto R0t2 =
+                m_parameters.get<ContactFrequency>().get_cont_freq_mat().get_matrix_at(m_result.get_last_time())(0, 0) *
+                m_parameters.get<TransmissionRisk>() * m_parameters.get<InfectiousTime>();
```

Comment:
Did you forget the TransmissionRisk in the first commits? Is this correct now?

Reply:
In the commits at the beginning, I defined the contacts as effective contacts and wrote this in the comments.
Now I separated the contacts and the transmission risk so that both versions were actually correct.